beacon-chain/db/kv/blocks.go

```diff
@@ -262,28 +262,34 @@ func (s *Store) SaveBlocks(ctx context.Context, blocks []block.SignedBeaconBlock
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveBlocks")
 	defer span.End()
 
+	blockRoots := make([][]byte, len(blocks))
+	encodedBlocks := make([][]byte, len(blocks))
+	indicesForBlocks := make([]map[string][]byte, len(blocks))
+	for i, blk := range blocks {
```

Comment:
maybe just add a small comment stating you are doing the expensive ops beforehand so that you do not hold the bolt lock longer than you have to.